tutorials/scripting/gdextension/gdextension_cpp_example.rst

```diff
@@ -0,0 +1,652 @@
+.. _doc_gdextension_cpp_example:
+
+GDExtension C++ example
+=======================
+
+Introduction
+------------
+
+The C++ bindings for GDExtension are built on top of the C GDExtension API
+and provide a nicer way to "extend" nodes in Godot using C++.
+This new system allows the extension of Godot to nearly the same
+level as statically linked C++ modules.
+
+You can download the included example in the test folder of the godot-cpp
+repository `on GitHub <https://github.com/godotengine/godot-cpp>`__.
+
+Setting up the project
+----------------------
+
+There are a few prerequisites you'll need:
+
+- a Godot 4 executable,
+- a C++ compiler,
+- SCons as a build tool,
+- a copy of the `godot-cpp
+  repository <https://github.com/godotengine/godot-cpp>`__.
+
+See also :ref:`Compiling <toc-devel-compiling>` as the build tools are identical
+to the ones you need to compile Godot from source.
+
+You can download this repository from GitHub or let Git do the work for you.
+Note that this repository has different branches for different versions
+of Godot. GDExtensions will not work in older versions of Godot (only Godot 4 and up) and vice versa, so make sure you download the correct branch.
+
+.. note::
+
+    `GDExtension <https://godotengine.org/article/introducing-gd-extensions>`__
+    has been merged in the ``master`` branch of godot-cpp,
+    but it is only compatible with the upcoming Godot 4.0.
+    Therefore, you need to use the ``master`` branch of godot-cpp to use GDExtension
+    and follow this example.
+
+If you are versioning your project using Git, it is a good idea to add them as
+Git submodules:
+
+.. code-block:: none
+
+    mkdir gdextension_cpp_example
+    cd gdextension_cpp_example
+    git init
+    git submodule add -b master https://github.com/godotengine/godot-cpp
+    cd godot-cpp
+    git submodule update --init
+
+If you decide to just download the repositories or clone them into your project
+folder, make sure to keep the folder layout identical to the one described here,
+as much of the code we'll be showcasing here assumes the project follows this
+layout.
+
+Do make sure you clone recursively to pull in both repositories:
+
+.. code-block:: none
+
+    mkdir gdextension_cpp_example
+    cd gdextension_cpp_example
+    git clone --recursive -b master https://github.com/godotengine/godot-cpp
+
+.. note::
+
+    ``godot-cpp`` now includes ``godot-headers`` as a nested submodule, if you've
```

Comment:
Need to double check this, I believe that in GDExtensions we no longer maintain the `godot-headers` repo and just maintain all files in `godot-cpp`.

I think you can thus also remove the mention of needing to recursively clone.

Might want to double check this but I think this is so.

Reply:
You are right about this. I will remove this then